### PR TITLE
OAuth2 single use token (expires after hour, requires Xenia restart) (#19)

### PR DESCRIPTION
**Pull request by zbyszkop**
Adds oauth2 authorization, without refresh yet. (#19 )

### CODE REVIEW

File: xenia-api/src/main/groovy/pl/jug/torun/xenia/meetup/MeetupClient.groovy [hunk 1]
```diff
@@ -1,31 +1,56 @@
...
+    }
+
+    private HttpEntity getAuthHeaders() {
+        if (oAuthData.requestToken == null) {
+            throw new RuntimeException("Access token not set!")
+        }
+
+        HttpHeaders headers = new HttpHeaders();
+        headers.set("Authorization", "Bearer " + oAuthData.requestToken)
+        HttpEntity entity = new HttpEntity(headers)
```
Comment (zbyszkop) [inline xenia-api/src/main/groovy/pl/jug/torun/xenia/meetup/MeetupClient.groovy:52]: This variable creation is unnecessary.

File: xenia-api/src/main/groovy/pl/jug/torun/xenia/meetup/MeetupClient.groovy [hunk 2]
```diff
@@ -1,31 +1,56 @@
...
+import pl.jug.torun.xenia.oauth.OAuthData
+
+
 
 @Component
 final class MeetupClient {
 
     private final MeetupRestTemplate restTemplate
 
+    @Autowired
```
Comment (kapware) [inline xenia-api/src/main/groovy/pl/jug/torun/xenia/meetup/MeetupClient.groovy:20]: You don't have to use `@Autowired` at all, consider following replacement:

```java
private final OAuthData oAuthData;

MeetupClient(MeetupRestTemplate restTemplate, OAuthData oAuthData) {
         this.restTemplate = restTemplate
         this.oAuthData = oAuthData
}
```

File: xenia-api/src/main/groovy/pl/jug/torun/xenia/oauth/OAuthData.groovy [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    String clientSecret
+
+    @PostConstruct
+    void init() {
+        String address = "https://secure.meetup.com/oauth2/authorize" +
+                "?client_id="+ clientId +
+                "&response_type=code" +
+                "&redirect_uri=http://localhost:8080/oauth2/redirect";
+
+        if (Desktop.isDesktopSupported() && Desktop.getDesktop().isSupported(Desktop.Action.BROWSE)) {
```
Comment (kapware) [inline xenia-api/src/main/groovy/pl/jug/torun/xenia/oauth/OAuthData.groovy:28]: I understand the shortcut, but this is rather bad UX, I think exposing this on a separate endpoint would be beneficial. Then UI can slurp it, redirect and after successful access can redirect back to app seamlessly.
This scenario is on my TODO list.

**Review by kapware — COMMENTED** [review]
Well, this works :) But I'd like to try another approach with a slightly better UX, before we merge this in.
I'll let it rot here for a while, unless there's a better solution and meetup disables keys we'll merge this in.
Thanks Zbyszko!

**Comment by zbyszkop** [comment]
@kapware thanks for review:) 
Yeah, UX wasn't my prime concern because of the urgency of the issue. Now that a working solution exists, I think I'll find some time to create a better UX for this, without the time pressure. We could actually pair up on this, if you want.

**Pull request merged**